components/eam/src/physics/cam/hetfrz_classnuc.F90

```diff
@@ -310,272 +249,503 @@ subroutine hetfrz_classnuc_calc( &
    frzdudep = 0._r8
 
    ! critical germ size
-   rgimm = 2*vwice*sigma_iw/(kboltz*t*LOG(supersatice))
+   rgimm = 2*vwice*sigma_iw/(kboltz*temperature*log(supersatice))
+   
    ! take solute effect into account
    rgimm_bc = rgimm
    rgimm_dust_a1 = rgimm
    rgimm_dust_a3 = rgimm
 
-   ! if aw*Si<=1, the freezing point depression is strong enough to prevent freezing
+   call calculate_rgimm_and_determine_spec_flag(vwice, sigma_iw, temperature, aw(id_bc), supersatice, & ! in
```

Comment:
Here, should the dummy argument for the 'rgimm_*' variables be intent(inout), just like in the  'calculate_water_activity' case?  Otherwise , the initialization values will become undefined if  aw Si <=1.

Reply:
changed variable rgimm_xx to be inout.